cpp/frameProcessor/include/FrameProcessorPlugin.h

```diff
@@ -29,10 +29,11 @@ struct JsonNullType{};
 */
 struct ParamMetadata
 {
+  using param_metadata_vec = std::vector<boost::variant<std::string, int>>;
   const std::string path;
   const std::string type;
   const std::string access_mode;
-  const std::vector<boost::variant<std::string, int>> allowed_vals;
+  const param_metadata_vec allowed_values;
```

Comment:
This name is very confusing. It is not a vector of parameter metadata, that would better describe what the plugins will return from `get_*_metadata`. What about this?

```suggestion
  using allowed_value_t = boost::variant<std::string, int>;
  const std::vector<allowed_value_t> allowed_values;
```

Is this type going to be needed by the specific plugins?